Framework/Kernel/src/ConfigService.cpp

```diff
@@ -1802,37 +1810,59 @@ void ConfigServiceImpl::updateFacilities(const std::string &fName) {
+  bool success = false;
+  while ((!success) && (attemptIndex < fileNames.size())) {
+    std::string fileName = fileNames[attemptIndex];
+    try {
+      // Set up the DOM parser and parse xml file
+      Poco::AutoPtr<Poco::XML::Document> pDoc;
+      try {
+        Poco::XML::DOMParser pParser;
+        pDoc = pParser.parse(fileName);
+      } catch (...) {
+        throw Kernel::Exception::FileError("Unable to parse file:", fileName);
+      }
 
-  // Set up the DOM parser and parse xml file
-  Poco::AutoPtr<Poco::XML::Document> pDoc;
-  try {
-    Poco::XML::DOMParser pParser;
-    pDoc = pParser.parse(fileName);
-  } catch (...) {
-    throw Kernel::Exception::FileError("Unable to parse file:", fileName);
-  }
+      // Get pointer to root element
+      Poco::XML::Element *pRootElem = pDoc->documentElement();
+      if (!pRootElem->hasChildNodes()) {
+        throw std::runtime_error("No root element in Facilities.xml file");
+      }
 
-  // Get pointer to root element
-  Poco::XML::Element *pRootElem = pDoc->documentElement();
-  if (!pRootElem->hasChildNodes()) {
-    throw std::runtime_error("No root element in Facilities.xml file");
-  }
+      Poco::AutoPtr<Poco::XML::NodeList> pNL_facility =
+          pRootElem->getElementsByTagName("facility");
+      size_t n = pNL_facility->length();
 
-  Poco::AutoPtr<Poco::XML::NodeList> pNL_facility =
-      pRootElem->getElementsByTagName("facility");
-  size_t n = pNL_facility->length();
+      for (unsigned long i = 0; i < n; ++i) {
+        auto *elem = dynamic_cast<Poco::XML::Element *>(pNL_facility->item(i));
+        if (elem) {
+          m_facilities.push_back(new FacilityInfo(elem));
+        }
+      }
 
-  for (unsigned long i = 0; i < n; ++i) {
-    auto *elem = dynamic_cast<Poco::XML::Element *>(pNL_facility->item(i));
-    if (elem) {
-      m_facilities.push_back(new FacilityInfo(elem));
-    }
-  }
+      if (m_facilities.empty()) {
+        throw std::runtime_error("The facility definition file " + fileName +
+                                 " defines no facilities");
+      }
 
-  if (m_facilities.empty()) {
-    throw std::runtime_error("The facility definition file " + fileName +
-                             " defines no facilities");
+      // if we got here we have suceeded and can exit the loop
+      success = true;
+    } catch (std::runtime_error &ex) {
+      // log this failure to load a file
+      g_log.error() << "Failed to load the facilities.xml file at " << fileName
+                    << "\nIt might be corrupt.  " << ex.what()
+                    << "\nWill try to load another version.\n";
+      attemptIndex++;
+      // move on to the next file index if available
+      if (attemptIndex == fileNames.size()) {
+        // Throw an exception as we have run out of files to try
+        throw std::runtime_error(
+            "No more Facilities.xml files can be found, Mantid will not be "
+            "able to start, Sorry.  Try reinstalling Mantid.");
```

Comment:
Since this error seems to be swallowed by the error reporter, it may be worth explicitly sending this message to the log.